src/Transport/Receiving/MessagePump.cs

```diff
@@ -243,6 +243,10 @@ async Task ProcessMessage(Task<Message> receiveTask)
                         await receiver.SafeAbandonAsync(pushSettings.RequiredTransactionMode, lockToken).ConfigureAwait(false);
                     }
                 }
+                catch (Exception onErrorException) when (onErrorException is MessageLockLostException || onErrorException is ServiceBusTimeoutException)
+                {
+                    logger.Debug("Failed to execute recoverability.", onErrorException);
```

Comment:
what happens with the message here given that this path won't call `SafeAbandonAsync`?

Reply:
The (reference) to the message is useless to us at this point as we no longer have a valid lock or there's a timeout error, indicating we cannot communicate. The message on the broker is already consumed by another thread/consumer at this point.

Reply:
👍 makes sense, thanks for clarifying this. Might have been a useful comment in the code for future readers.